pybamm/expression_tree/unary_operators.py

```diff
@@ -901,6 +901,9 @@ def x_average(symbol):
     :class:`Symbol`
         the new averaged symbol
     """
+    # Can't take average if the symbol evaluates on edges
```

Comment:
why can't we average on edges? is it just that it isn't implemented? (if so, no need to do this now. it can be a separate issue if/when we need it)

Reply:
No reason why we can't in theory. I'm not sure what the proper implementation would be in a finite volume context though? For nodes it's clearly just sum all the values, but for edges I don't know what it should be: maybe `N[-1] - N[0]`?